lib/src/widgets/yaru_page_indicator.dart

```diff
@@ -48,8 +56,17 @@ class YaruPageIndicator extends StatelessWidget {
   /// Will be automatically reduced to fit the vertical constraints.
   final double dotSpacing;
 
+  /// Decoration of the dots.
+  final YaruDotDecorationBuilder? dotDecorationBuilder;
+
   @override
   Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final indicatorTheme = YaruPageIndicatorTheme.of(context);
+
+    final dotSize = indicatorTheme?.dotSize ?? this.dotSize;
+    final dotSpacing = indicatorTheme?.dotSpacing ?? this.dotSpacing;
```

Comment:
we need to swap these around. a local widget property should win over a theme-level property. perhaps we need to make the widget properties nullable?

Reply:
Oh, right.
It's a bit of a stupid mistake from myself to invert that 😅

I'll also make the widget properties nullable :+1: